sdk/identity/azure-identity/test/ut/client_secret_credential.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/identity/client_secret_credential.hpp"
+
+#include <azure/core/io/body_stream.hpp>
+
+#include "test_transport.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace Azure::Identity;
+
+namespace {
+
+struct CredentialResult
+{
+  struct RequestInfo
+  {
+    std::string AbsoluteUrl;
+    Azure::Core::CaseInsensitiveMap Headers;
+    std::string Body;
+  } Request;
+
+  struct
+  {
+    std::chrono::system_clock::time_point Earliest;
+    std::chrono::system_clock::time_point Latest;
+    Azure::Core::Credentials::AccessToken AccessToken;
+  } Response;
+};
+
+CredentialResult TestClientSecretCredential(
+    std::string const& tenantId,
+    std::string const& clientId,
+    std::string const& clientSecret,
+    ClientSecretCredentialOptions credentialOptions,
+    Azure::Core::Credentials::TokenRequestContext const& tokenRequestContext,
+    std::string const& responseBody)
+{
+  CredentialResult result;
+
+  auto responseVec = std::vector<uint8_t>(responseBody.begin(), responseBody.end());
+  credentialOptions.Transport.Transport = std::make_shared<TestTransport>([&](auto request, auto) {
+    auto const bodyVec = request.GetBodyStream()->ReadToEnd(Azure::Core::Context());
+
+    result.Request
+        = {request.GetUrl().GetAbsoluteUrl(),
+           request.GetHeaders(),
+           std::string(bodyVec.begin(), bodyVec.end())};
+
+    auto response = std::make_unique<Azure::Core::Http::RawResponse>(
+        1, 1, Azure::Core::Http::HttpStatusCode::Ok, "OK");
+
+    response->SetBodyStream(std::make_unique<Azure::Core::IO::MemoryBodyStream>(responseVec));
+
+    result.Response.Earliest = std::chrono::system_clock::now();
+    return response;
+  });
+
+  ClientSecretCredential credential(tenantId, clientId, clientSecret, credentialOptions);
+  result.Response.AccessToken = credential.GetToken(tokenRequestContext, Azure::Core::Context());
+  result.Response.Latest = std::chrono::system_clock::now();
+
+  return result;
+}
+} // namespace
+
+TEST(ClientSecretCredential, Regular)
```

Comment:
How can I see the test coverage for this change and azure-identity overall?

It's not visible here: https://dev.azure.com/azure-sdk/public/_build/results?buildId=808213&view=results

Reply:
I don't know how the code coverage integration works, but if you don't see it, it most likely was not enable for identity. We need a work item to enable it.